engine/inputdata/metaparameter.ftl

```diff
@@ -0,0 +1,188 @@
+            "Type" : STRING_TYPE,
+            "Mandatory" : true
+        },
+        {
+            "Names" : ["Function"],
+            "Type" : STRING_TYPE
+        },
+        {
+            "Names" : ["Service"],
+            "Type" : STRING_TYPE
+        },
+        {
+            "Names" : ["Task"],
+            "Type" : STRING_TYPE
+        },
+        {
+            "Names" : ["PortMapping", "Port"],
+            "Type" : STRING_TYPE
+        },
+        {
+            "Names" : ["Mount"],
+            "Type" : STRING_TYPE
+        },
+        {
+            "Names" : ["Platform"],
+            "Type" : STRING_TYPE
+        },
+        {
+            "Names" : [ "RouteTable" ],
+            "Type" : STRING_TYPE
+        },
+        {
+            "Names" : [ "NetworkACL" ],
+            "Type" : STRING_TYPE
+        },
+        {
+            "Names" : [ "DataBucket" ],
+            "Type" : STRING_TYPE
+        },
+        {
+            "Names" : [ "Key" ],
+            "Type" : STRING_TYPE
+        },
+        {
+            "Names" : [ "Branch" ],
+            "Type" : STRING_TYPE
+        },
+        {
+            "Names" : [ "Client" ],
+            "Type" : STRING_TYPE
+        },
+        {
+            "Names" : [ "Connection" ],
+            "Type" : STRING_TYPE
+        },
+        {
+            "Names" : [ "AuthProvider" ],
+            "Type" : STRING_TYPE
+        },
+        {
+            "Names" : [ "Resource" ],
+            "Type" : STRING_TYPE
+        },
+        {
+            "Names" : [ "DataFeed" ],
+            "Type" : STRING_TYPE
+        },
+        {
+            "Names" : [ "RegistryService" ],
+            "Type" : STRING_TYPE
+        },
+        {
+            "Names" : [ "Assignment" ],
+            "Type" : STRING_TYPE
+        },
+        {
+            "Names" : [ "Route" ],
+            "Type"  : STRING_TYPE
+        },
+        {
+            "Names" : [ "Endpoint" ],
+            "Type"  : STRING_TYPE
+        },
+        {
+            "Names" : "Instance",
+            "Type" : STRING_TYPE
+        },
+        {
+            "Names" : "Version",
+            "Type" : STRING_TYPE
+        },
+        {
+            "Names" : "Secret",
+            "Type" : STRING_TYPE
+        }
+    ]
+]
+
+[#assign
+    linkChildrenConfiguration =
+        filterChildrenConfiguration +
+        [
+            {
+                "Names" : "Role",
+                "Type" : STRING_TYPE
+            },
+            {
+                "Names" : "Direction",
+                "Type" : STRING_TYPE
+            },
+            {
+                "Names" : "Type",
+                "Type" : STRING_TYPE
+            },
+            {
+                "Names" : "Enabled",
+                "Type" : BOOLEAN_TYPE,
+                "Default" : true
+            },
+            {
+                "Names" : "IncludeInContext",
+                "Type" : ARRAY_OF_STRING_TYPE
+            }
+        ]
+]
+
```

Comment:
This should probably be part of the shared provider rather than in the engine. The engine is more responsible for implementing the storage and I/O for the inputdata type rather than the data that is injected.

Reply:
Had a good discussion with @roleyfoley regarding this. It's difficult to move these assignments into `metaparameters/links/id.ftl` because of the load-order of configuration. Messing with the load order causes other complications.

This led to a discussion around how metaparameters should be referenced in the first place - instead of being assigned as variables like this and referenced in many places, Metaparameters more directly align with `AttributeSets` - its a collection of attributes that is itself a composite object structure, however it can be used in many places and "referenced" by components or other types (blueprints etc). This would solve issues relating to load-order because a reference value that is just a string ie. `{ "Ref" : "links" }` is resolvable as files are loading in, but then during composite object processing (which is all post-load of freemarker files) this can resolve to an actual full schema where "ref" values are replaced with `{"Children": [ <insert-links-composite-object> ]}`. 

This also perfectly aligns with JSONSchema "ref" attribute which is starting to become [difficult to use due to its implementation only within JSONSchema](https://github.com/hamlet-io/engine/issues/1477) 

I have moved `linkChildConfiguration` back to its original location in `component.ftl` and will log a new ticket to implement the "Ref" attribute within `getCompositeObject` and rename `Metaparameters` to `AttributeSets`.